include/openenclave/internal/str.h

```diff
@@ -424,6 +424,78 @@ MEM_INLINE int str_split(str_t* str, const char* delim, str_t* lhs, str_t* rhs)
     return 0;
 }
 
+/**
+ * This type contains a Universally Unique Resource Identifier (UUID) type as
+ * defined in RFC4122. These UUID values are used to identify Trusted
+ * Applications.
+ */
+typedef struct
+{
+    uint32_t timeLow;
+    uint16_t timeMid;
+    uint16_t timeHiAndVersion;
+    uint8_t clockSeqAndNode[8];
+} TEEC_UUID;
+
+MEM_INLINE int uuid_from_string(str_t* str, TEEC_UUID* uuid)
+{
+    int i;
+    uint64_t uuid_parts[5];
+    char* id_copy;
+    char* posn;
+    char* current_token;
+
+#if defined(__linux__)
+    id_copy = strdup(str_ptr(str));
+#else
+    id_copy = _strdup(str_ptr(str));
+#endif
```

Comment:
These ifdefs are not needed, just use _strdup .  See host.h which #defines that as strdup on linux.  (Or in theory that could be swapped in host.h to just use strdup which is the [ISO standard](https://en.cppreference.com/w/c/experimental/dynamic/strdup) and define strdup as _strdup on Windows.)

Reply:
Good point. Updated.